auth-web/src/components/pay/eft/ShortNameRefund.vue

```diff
@@ -25,46 +28,266 @@
+      default: () => ({
+        shortName: null
+      })
     },
     unsettledAmount: {
       type: String,
       default: ''
     }
   },
   setup (props, { root }) {
+    const userStore = useUserStore()
     const state = reactive({
+      declineReason: '',
       actionDropdown: [],
       isShortNameLinkingDialogOpen: false,
       eftShortNameSummary: {},
       results: [],
-      totalResults: 0,
+      totalResults: 1,
       filters: {
         pageNumber: 1,
         pageLimit: 5
       },
       loading: false,
       options: _.cloneDeep(DEFAULT_DATA_OPTIONS),
-      expanded: []
+      expanded: [],
+      currentEftRefund: null,
+      isEftRefundApprover: computed(() => userStore.currentUser.roles.includes(Role.EftRefundApprover))
     })
+    const confirmationDialog: Ref<InstanceType<typeof ModalDialog>> = ref(null)
+    const headers = [
+      {
+        col: 'createdName',
+        hasFilter: false,
+        width: '300px',
+        value: 'Initiated by'
+      },
+      {
+        col: 'casSupplierNumber',
+        hasFilter: false,
+        width: '210px',
+        value: 'Supplier Record Number'
+      },
+      {
+        col: 'refundAmount',
+        hasFilter: false,
+        width: '260px',
+        value: 'Refund Amount'
+      },
+      {
+        col: 'actions',
+        hasFilter: false,
+        value: 'Actions',
+        width: '200px'
+      }
+    ]
 
-    const isLinked = computed<boolean>(() => {
-      return state.totalResults > 0 || state.loading
-    })
+    async function approveRefund (item) {
+      const shortNameRefund = {
+        status: EFTRefundType.APPROVED
+      }
+      try {
+        await PaymentService.patchEFTRefund(item.id, shortNameRefund)
+        await loadTransactions(props.shortNameDetails.id)
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to approve refund.', error)
+      }
+    }
+
+    function declineRefund (item) {
+      state.currentEftRefund = item
+      confirmationDialog.value.open()
+    }
+
+    async function dialogDecline () {
+      const shortNameRefund = {
+        status: EFTRefundType.DECLINED,
+        decline_reason: state.declineReason
+      }
+      try {
+        await PaymentService.patchEFTRefund(state.currentEftRefund.id, shortNameRefund)
+        await loadTransactions(props.shortNameDetails.id)
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to decline refund.', error)
+      }
+      confirmationDialog.value.close()
+      state.currentEftRefund = null
+    }
+
+    watch(() => props.shortNameDetails, () => {
+      return loadTransactions(props.shortNameDetails.id)
+    }, { immediate: true, deep: true })
+
+    async function loadTransactions (shortnameId: string): Promise<void> {
+      try {
+        state.loading = true
+        const response = await PaymentService.getEFTShortnameHistory(shortnameId, state.filters)
```

Comment:
@rodrigo-barraza , should this be using eft refund search instead of the short name history route?

Reply:
I tried creating an eft refund and it doesn't show up in the short name refund component. Still empty.
![Screenshot 2024-10-16 at 2 14 04 PM](https://github.com/user-attachments/assets/e33cd01e-4665-480d-ad09-0775b03cbbc8)